src/Package/Fetch.zig

```diff
@@ -1110,7 +1110,23 @@ fn unpackResource(
             try unpackTarball(f, tmp_directory.handle, dcp.reader());
         },
         .@"tar.xz" => try unpackTarballCompressed(f, tmp_directory.handle, resource, std.compress.xz),
-        .@"tar.zst" => try unpackTarballCompressed(f, tmp_directory.handle, resource, ZstdWrapper),
+        .@"tar.zst" => {
+            const gpa = f.arena.child_allocator;
+            const reader = resource.reader();
+            var br = std.io.bufferedReaderSize(std.crypto.tls.max_ciphertext_record_len, reader);
+
+            const window_size = std.compress.zstd.DecompressStreamOptions.default_window_size_max;
+            var window_buffer = gpa.create([window_size]u8) catch |err| {
+                return f.fail(f.location_tok, try eb.printString(
+                    "unable to decompress tarball: {s}",
+                    .{@errorName(err)},
+                ));
+            };
+            defer gpa.destory(window_buffer);
+
+            var decompress = std.compress.zstandard.decompress(br.reader(), &window_buffer);
+            try unpackTarball(f, tmp_directory.handle, decompress.reader());
+        },
```

Comment:
This looks a bit ugly given the almost identical code in `unpackTarballCompressed` - perhaps `unpackTarballCompressed` should rather have a buffer passed in to it.